### PR TITLE
Python: Add user agent to OpenAI and OpenAPI headers

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Motivation and Context

Replaces: https://github.com/microsoft/semantic-kernel/pull/3074

This PR adds the User-Agent header "Semantic Kernel" to calls made to OpenAI/Azure OpenAI and OpenAPI calls. 

### Description

* Addition of telemetry.py file containing telemetry constants
* ability to disable Azure/OpenAI telemetry with an environment variable called AZURE_TELEMETRY_DISABLED

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/connectors/telemetry.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import os
+
+import pkg_resources
+
+TELEMETRY_DISABLED_ENV_VAR = "AZURE_TELEMETRY_DISABLED"
+
+IS_TELEMETRY_ENABLED = (
+    os.environ.get(TELEMETRY_DISABLED_ENV_VAR, "false").lower() != "true"
```
Comment (dluc) [inline python/semantic_kernel/connectors/telemetry.py:10]: for consistency with the C#/Azure logic, this should check also for "1"

Comment (dluc) [reply python/semantic_kernel/connectors/telemetry.py:10]:
```suggestion
IS_TELEMETRY_ENABLED = os.environ.get(
    TELEMETRY_DISABLED_ENV_VAR, "false"
).lower() not in ["true", "1"]
```

File: python/semantic_kernel/connectors/telemetry.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+)
```
Comment (dluc) [inline python/semantic_kernel/connectors/telemetry.py:11]:
```suggestion
IS_TELEMETRY_ENABLED = os.environ.get(
    TELEMETRY_DISABLED_ENV_VAR, "false"
).lower() not in ["true", "1"]
```

File: python/semantic_kernel/connectors/telemetry.py (hunk 1 above)
Comment (markwallace-microsoft) [reply python/semantic_kernel/connectors/telemetry.py]: Applied your suggestion

File: python/semantic_kernel/connectors/telemetry.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+HTTP_USER_AGENT = "Semantic-Kernel"
+
+APP_INFO = {
+    "name": HTTP_USER_AGENT if IS_TELEMETRY_ENABLED else "",
+    "version": pkg_resources.get_distribution("semantic-kernel").version
+    if IS_TELEMETRY_ENABLED
+    else "",
+    "url": "",
+}
```
Comment (dluc) [inline python/semantic_kernel/connectors/telemetry.py:21]:
```suggestion
APP_INFO = (
    {
        "name": HTTP_USER_AGENT,
        "version": pkg_resources.get_distribution("semantic-kernel").version,
        "url": "",
    }
    if IS_TELEMETRY_ENABLED
    else None
)
```

this should avoid appending an empty space after the user agent when the telemetry is disabled (User-Agent: `OpenAI/v1 PythonBindings/0.28.0 ` vs User-Agent: `OpenAI/v1 PythonBindings/0.28.0`)

**Comment by dluc** [comment]
I would check the integration tests before merging. Something merged previously probably, I get at least one failure:

> poetry run pytest tests/integration/completions/test_azure_oai_text_service.py

> ERROR    root:test_utils.py:20 Retry 1: Error: 'AzureTextCompletion' object has no attribute '_prompt_tokens'

**Review by alexchaomander — APPROVED** [review]
LGTM!

**Review by lemillermicrosoft — APPROVED** [review]

**Review by awharrison-28 — APPROVED** [review]

**Pull request merged**